### PR TITLE
feat(lint/noConstantCondition): allow `while(true)`

### PR DESCRIPTION
**Pull request by Conaclos**
## Summary

This implements what we discuss in [rome#4746](https://github.com/rome/tools/discussions/4746).
`while (true) {}` is now ignored by `noConstantCondition`.

## Test Plan

Updated tests.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -325,6 +325,19 @@ The following rules are now recommended:
...
+  The rule now allows `while(true)`.
+  This recognizes a common pattern in the web community:
+
+  ```js
+  while (true) {
+    if (cond) {
+      break;
+    }
+  }
+  ```
```
Comment (ematipico) [inline CHANGELOG.md:339]: Shouldn't we track the code action too?

Comment (Conaclos) [reply CHANGELOG.md:339]: What do you mean?

Comment (ematipico) [reply CHANGELOG.md:339]: I mean the rule now has a code action, and the changelog doesn't say anything about it

Comment (Conaclos) [reply CHANGELOG.md:339]: This is a mistake. This should be part of another PR. Thanks for bringing it to light!

File: crates/rome_js_analyze/src/semantic_analyzers/nursery/no_constant_condition.rs [hunk 2]
```diff
@@ -106,6 +112,16 @@ impl Rule for NoConstantCondition {
...
         let test = conditional_stmt.test()?;
+        // Ignore `while (true) { ... }`
+        if let ConditionalStatement::JsWhileStatement(_) = conditional_stmt {
+            if let Some(test) = test.as_any_js_literal_expression() {
+                if let Some(test) = test.as_js_boolean_literal_expression() {
+                    if test.value_token().ok()?.kind() == JsSyntaxKind::TRUE_KW {
+                        return None;
+                    }
+                }
+            }
```
Comment (ematipico) [inline crates/rome_js_analyze/src/semantic_analyzers/nursery/no_constant_condition.rs:123]: You can chain all these and avoid at least two `if let Some` by using `.and_then`

File: crates/rome_js_analyze/src/semantic_analyzers/nursery/no_constant_condition.rs [hunk 3]
```diff
@@ -263,7 +279,7 @@ fn is_constant_condition(
         }
         JsIdentifierExpression(node) => {
             if node.name().ok()?.binding(model).is_some() {
-                // This is any_js_stmt edge case. Mordern browser doesn't allow to redeclare `undefined` but ESLint handle this so we do
+                // This is any_js_stmt edge case. Modern browser doesn't allow to redeclare `undefined` but ESLint handle this so we do
```
Comment (ematipico) [inline crates/rome_js_analyze/src/semantic_analyzers/nursery/no_constant_condition.rs:282]:
```suggestion
                // This is any_js_stmt edge case. Modern browsers don't allow to redeclare `undefined` but ESLint handle this so we do
```

**Pull request merged**